src/kirby/components/form-field/input/input.component.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  HostBinding,
+  HostListener,
+  Output,
+  EventEmitter,
+} from '@angular/core';
+
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  // tslint:disable-next-line:component-selector
+  selector: 'input[kirby-input]',
+  styleUrls: ['./input.component.scss'],
+  template: '',
+})
+export class InputComponent {
+  @Output() change = new EventEmitter<string>();
+
+  private static typeToInputmodeMap = {
+    number: 'decimal',
+    search: 'search',
+  };
+
+  @Input() set type(value: string) {
+    this._inputmode = InputComponent.typeToInputmodeMap[value];
+  }
+
+  @HostBinding('class.error')
+  @Input()
+  hasError: boolean;
+
+  @HostBinding('attr.autocomplete')
+  @Input()
+  autocomplete: 'on' | 'off' = 'off';
+
+  @HostBinding('attr.autocorrect')
+  @Input()
+  autocorrect: 'on' | 'off' = 'off';
+
+  @HostBinding('attr.value')
+  @Input()
+  value: string;
+
+  @HostBinding('attr.maxlength')
+  @Input()
+  maxlength: number;
+
+  @HostBinding('attr.inputmode')
+  private _inputmode: string;
+
+  @HostListener('keyup', ['$event.target.value'])
+  private _onKeyUp(value: string) {
```

Comment:
Is there a reason you not just using the `oninput` event? see [here](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/oninput).

It also is triggered by cut/paste.